scripts/main.js

```diff
@@ -0,0 +1,348 @@
+"use strict";
+document.addEventListener("keydown", (event) => move(event.key));
+
+const rows = document.querySelectorAll("tr");
+
+const button = document.querySelector(".button");
+const score = document.querySelector(".game-score");
+
+let scoreVal = 0;
+
+const board = [
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+];
+
+const message_start = document.querySelector(".message-start");
+const message_lose = document.querySelector(".message-lose");
+const message_win = document.querySelector(".message-win");
+
+button.addEventListener("click", () => {
+  if (button.classList.contains("start")) {
+    start();
+  } else {
+    restart();
+  }
+});
+
+function start() {
+  button.innerHTML = "Restart";
+  button.classList.add("restart");
+  button.classList.remove("start");
+
+  message_start.classList.add("hidden");
+
+  newCell();
+  newCell();
+  drawBoard();
+}
+
+function restart() {
+  scoreVal = 0;
+  button.innerHTML = "Start";
+  button.classList.add("start");
+  button.classList.remove("restart");
+
+  message_win.classList.add("hidden");
+  message_lose.classList.add("hidden");
+  message_start.classList.remove("hidden");
+
+  for (let y = 0; y < 4; y++) {
+    for (let x = 0; x < 4; x++) {
+      board[y][x] = 0;
+    }
+  }
+  drawBoard();
+}
+
+function move(key) {
+  let canMove = false;
+  if (
+    message_start.classList.contains("hidden") &&
+    message_win.classList.contains("hidden") &&
+    message_lose.classList.contains("hidden")
+  ) {
+    if (key === "ArrowUp") {
+      canMove = moveUp();
+    } else if (key === "ArrowDown") {
+      canMove = moveDown();
+    } else if (key === "ArrowLeft") {
+      canMove = moveLeft();
+    } else if (key === "ArrowRight") {
+      canMove = moveRight();
+    }
+
+    if (canMove) {
+      setTimeout(() => {
+        newCell();
+        drawBoard();
+        if (board.find((row) => row.find((el) => el === 2048))) {
+          message_win.classList.remove("hidden");
+        }
+      }, 300);
+    }
+  }
```

Comment:
The 'move' function contains a lot of logic and directly manipulates the DOM. It would be beneficial to break this function down into smaller, more focused functions that handle specific tasks, such as updating the game state or checking for win/lose conditions.